megazords/ios-rust/build-xcframework.sh

```diff
@@ -179,4 +180,3 @@ rm -rf "$XCFRAMEWORK_ROOT/common"
 # Zip it all up into a bundle for distribution.
 
 (cd "$THIS_DIR" && zip -9 -r "$FRAMEWORK_NAME.xcframework.zip" "$FRAMEWORK_NAME.xcframework")
-rm -rf "$XCFRAMEWORK_ROOT"
```

Comment:
I'm curious why we delete this here, is there a reason we keep the xcframework directory after zipping it up? Is this for local debugging?

Reply:
Yeah this is one of the other areas up for discussion. Basically, to build all our tests, we need to somehow link our built rust libs to the tests so Xcode can compile it. This means two options:

1. Use `rust-components-swift`:
   + (plus) Easier to understand/use
   - (negative) To test your code, you'll need to make a local a-s xcframework, put it into `rust-components-swift` then change the package to point the locally
Ultimately this feels like a weird workaround and probably is worse than carthage for testing new code quickly.

2. Build Xcframework locally as part of the Xcode build process, and link to that to run the tests:
  - (plus) Don't have to think about it, part of Xcode build process
  - (plus) We already have this built out, just need to run the script
  - (negative) Extra artifacts hanging around locally

I think 2 is definitely the way to go here but open to alternatives. I eventually want to make a task that deals with having these artifacts and .zip files only made when necessary as well as associated scripts to clean these when people run the respective clean scripts

Reply:
I think the second option makes sense. Consolidating the rough parts of this conversion to the a-s local development process where possible will make it easier for us to make improvements in the future.